app/controllers/application_controller.rb

```diff
@@ -1,10 +1,79 @@
+require 'pry'
 require './config/environment'
 
 class ApplicationController < Sinatra::Base
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "evelynwaugh"
   end
 
-end
\ No newline at end of file
+  helpers do
+    def logged_in?
+      if session[:user_id]
+        true
+      else
+        false
+      end
+    end
+
+  end
+
+  get '/' do
+    erb :'home/index'
+  end
+
+  get '/login' do
+    if logged_in?
+      redirect to '/tweets'
+    end
+    erb :'home/login'
+  end
+
+  get '/signup' do
+    if logged_in?
+      redirect to '/tweets'
+    end
+    erb :'home/signup'
+  end
+
+  post '/signup' do
+    if params[:user][:username].empty? || params[:user][:password].empty? || params[:user][:email].empty?
+      redirect to '/signup'
+    elsif !params[:user][:email].include?("@") || !params[:user][:email].include?(".")
+      redirect to '/signup'
+    else
+      @user = User.create(params[:user])
```

Comment:
we don't need to set an instance variable for a post route that we won't use on the view.